docs/source/articles/implement-the-credit-card-tender-type.html.md

```diff
@@ -0,0 +1,298 @@
+          merchant_account_id: 'a',
+          merchant_id:         'b',
+          public_key:          'c',
+          private_key:         'd',
+          environment:         'sandbox'
+        )
+    end
+
+    def reset_gateway
+      Workarea.config.gateways.credit_card = @_old_gateway
+    end
+  end
+end
+```
+
+For a usage example, refer to Workarea Braintree's
+[decoration of `Payment::CreditCardIntegrationTest`](https://github.com/workarea-commerce/workarea-braintree/blob/v1.0.3/test/models/workarea/payment/credit_card_integration_test.decorator):
+
+```ruby
+module Workarea
+  decorate Payment::CreditCardIntegrationTest, with: :braintree do
+    decorated { include BraintreeGatewayVCRConfig }
+  end
+end
+```
+
+The result of this and other decorations is a [collection of VCR cassettes](https://github.com/workarea-commerce/workarea-braintree/tree/v1.0.3/test/vcr_cassettes) that store the responses from the gateway to be used with each automated test run.
+
+
+Tokenization
+----------------------------------------------------------------------
+
+In addition to processing payments, a credit card gateway is responsible for card tokenization.
+Workarea encapsulates its tokenization logic within `Payment::StoreCreditCard`, particularly `Payment::StoreCreditCard#perform!` (with tests in `Payment::StoreCreditCardTest`).
+
+If your gateway's tokenization API differs from Workarea's default implementation, decorate `Payment::StoreCreditCard` to customize the implementation as necessary.
+The following snippets illustrate how the default bogus gateway and the Braintree gateway return the card token within a different parameter:
+
+```ruby
+# default implementation in Workarea Core
+module Workarea
+  class Payment
+    class StoreCreditCard
+      def perform!
+        # ...
+        @credit_card.token = response.params['billingid']
+      end
+    end
+  end
+end
+
+# extension by Workarea Braintree
+module Workarea
+  class Payment
+    class StoreCreditCard
+      def perform!
+        # ...
+        @credit_card.token = response.params['credit_card_token']
+      end
+    end
+  end
+end
+```
+
+Workarea Braintree also decorates tests to configure the correct gateway and account for gateway API differences.
+Refer to the following sources:
+
+* [Workarea Core implementation](https://github.com/workarea-commerce/workarea/blob/v3.4.18/core/app/models/workarea/payment/store_credit_card.rb)
+* [Workarea Core test case](https://github.com/workarea-commerce/workarea/blob/v3.4.18/core/test/models/workarea/payment/store_credit_card_test.rb)
+* [Workarea Braintree implementation extensions](https://github.com/workarea-commerce/workarea-braintree/blob/v1.0.3/app/models/workarea/payment/store_credit_card.decorator)
+* [Workarea Braintree test case extensions](https://github.com/workarea-commerce/workarea-braintree/blob/v1.0.3/test/models/workarea/payment/store_credit_card_test.decorator)
+
+
+Operation Implementations
```

Comment:
You should make reference to the #cancel method that must be implemented. This method handles rolling back payment operations on error. This method also rolls back payment of fraud for some of our fraud plugins.